include/xgboost/tree_model.h

```diff
@@ -613,6 +626,107 @@ inline void RegTree::CalculateContributions(const RegTree::FVec& feat, unsigned
   out_contribs[split_index] += leaf_value - node_value;
 }
 
+// used by _compute_weight
+inline unsigned _seq_prod(unsigned a, unsigned b) {
+  unsigned out = 1.0;
+  for (unsigned i = a; i <= b; ++i) out *= i;
+  return out;
+}
+
+inline bst_float _compute_weight(unsigned num_fixed, unsigned num_integrated) {
```

Comment:
Wouldn't it be faster to use a lookup table for a limited set of smaller argument values, and to use LogGamma computations for those not covered by the lookup table?

Also, this function naming style is out of place.